### PR TITLE
Fixes #3284 - Move submit issue logic to the refactored version

### LINKED ISSUE #3284: Move submit issue logic to the refactored version

### PR DESCRIPTION
**Pull request by ksy36**

### CODE REVIEW

File: webcompat/static/js/lib/wizard/steps/screenshot.js [hunk 1]
```diff
@@ -16,7 +21,7 @@ import {
 } from "../validation.js";
 import { showContainer } from "../ui-utils.js";
 import { downsampleImage, convertToDataURI } from "../utils.js";
-import uploadHelper from "./upload-helper/index.js";
```
Comment (ksy36) [inline webcompat/static/js/lib/wizard/steps/screenshot.js:19]: I've renamed these methods to distinguish between "saving" a image as a background and the actual upload that happens on form submit

File: webcompat/static/js/lib/wizard/steps/sub-category.js [hunk 2]
```diff
@@ -21,8 +21,19 @@ const resetRadio = (element) => {
...
+  const text = $(target).next("label").text().trim();
+  const toUpdate = {
+    data: { elementId: "#description", value: text },
+    single: true,
+  };
+  notify.publish("updateStep", { id: "hidden", data: toUpdate });
+};
+
+const handleSubCategory = (event) => {
+  updateDescription(event.currentTarget);
```
Comment (ksy36) [inline webcompat/static/js/lib/wizard/steps/sub-category.js:34]: I've noticed that after submitting subcategory is missing in the issue description, so this code is fixing it

File: webcompat/static/js/lib/wizard/utils.js [hunk 3]
```diff
@@ -74,3 +74,26 @@ export const isSelfReport = (href, origin) => {
...
+  backgroundImage, i.e, for the following two possible strings,
+  'url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAACAAAAAYACAIAAABt)'
+  'url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAACAAAAAYACAIAAABt")'
+  we expect "data:image/ping;base64,iVBORw0KGgoAAAANSUhEUgAACAAAAAYACAIAAABt"
+  to be matched.
+
+  Note: browsers are inconsistent in quoting CSSOM serialization
+*/
+export const getDataURIFromPreview = (bgImage) => {
+  const re = /url\(['"]{0,1}(data:image\/(?:jpeg*|jpg|png|gif|bmp);\s*base64,.+)['"]{0,1}\)/;
```
Comment (miketaylr) [inline webcompat/static/js/lib/wizard/utils.js:89]: I wonder if anyone has ever uploaded a `.jpe` image... I guess I could check on the server. Just wondering out loud.

File: webcompat/static/js/lib/wizard/steps/description.js [hunk 4]
```diff
@@ -40,11 +40,26 @@ const onChange = (value) => {
   updateProgress(percent);
 };
 
+/*
+  create the markdown with the URL of a newly uploaded image
+  and its thumbnail URL assets to the bug description
+*/
+const updateDescription = (url) => {
+  const imageURL =
+    `<details><summary>View the screenshot</summary>` +
```
[the remark this answers is not in the document]
Comment (ksy36) [reply webcompat/static/js/lib/wizard/steps/description.js]: ahh right, good point :)

File: webcompat/static/js/lib/wizard/steps/description.js [hunk 5]
```diff
@@ -40,11 +40,26 @@ const onChange = (value) => {
   updateProgress(percent);
 };
 
+/*
+  create the markdown with the URL of a newly uploaded image
+  and its thumbnail URL assets to the bug description
```
[the remark this answers is not in the document]
Comment (ksy36) [reply webcompat/static/js/lib/wizard/steps/description.js]: yup, changed it 👍

File: webcompat/static/js/lib/wizard/steps/upload-helper/image-upload.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+const previewEl = $(".js-image-upload");
+
+const addImageURL = ({ url }) => {
+  if (!url) return;
+  notify.publish("updateStep", { id: "description", data: { url } });
+};
+
+const handleUploadError = function (response) {
+  if (response && response.status === 415) {
+    wcEvents.trigger("flash:error", {
```
[the remark this answers is not in the document]
Comment (ksy36) [reply webcompat/static/js/lib/wizard/steps/upload-helper/image-upload.js:24]: yeah, good idea

**Pull request merged**